source/ed/ed2nav.cpp

```diff
@@ -165,6 +165,31 @@ struct FindAdminWithCities {
     }
 };
 
+bool write_data_to_file(const std::string& output_filename, navitia::type::Data& data) {
+    auto logger = log4cplus::Logger::getInstance("log");
+    std::string temp_output = output_filename + ".temp";
+    try {
+        data.save(temp_output);
+        if (remove(output_filename.c_str()) == 0) {
+            if (rename(temp_output.c_str(), output_filename.c_str()) == 0)
+                LOG4CPLUS_INFO(logger, "Data saved");
+            else
+                LOG4CPLUS_ERROR(logger, "Unable to rename the new data file");
+        } else {
+            LOG4CPLUS_ERROR(logger, "Error deleting file: No such file or directory");
+        }
+    } catch (const navitia::exception& e) {
+        LOG4CPLUS_ERROR(logger, "Unable to save");
+        LOG4CPLUS_ERROR(logger, e.what());
+        if (remove(temp_output.c_str()) == 0)
+            LOG4CPLUS_INFO(logger, "Temp data removed because it was corrupted");
+        else
+            LOG4CPLUS_ERROR(logger, "Error deleting temp file: No such file or directory");
```

Comment:
you log an error that may or may not be the one that occured, it will mislead people if the error is caused by something else like a filesystem in readonly. You should use `strerror` to get the correct error message